mode/src/processing/mode/android/SDKDownloader.java

```diff
@@ -603,17 +599,35 @@ public static String humanReadableByteCount(long bytes, boolean si) {
   public SDKDownloader(Frame editor) {
     super(editor, AndroidMode.getTextString("sdk_downloader.download_title"), true);
     this.editor = editor;
-    this.sdk = null;    
-    createLayout();
+    this.sdk = null;
+    downloadTask = new SDKDownloadTask();
+    SDKUrlHolder downloadUrls = new SDKUrlHolder();
+    String repositoryUrl = REPOSITORY_URL + REPOSITORY_LIST;
+    String addonUrl = REPOSITORY_URL + ADDON_LIST;
+    String haxmUrl = HAXM_URL + ADDON_LIST;
+    try {
+      downloadTask.getMainDownloadUrls(downloadUrls, repositoryUrl, Platform.getName());
+      getExtrasDownloadUrls(downloadUrls, addonUrl, Platform.getName());
+      getHaxmDownloadUrl(downloadUrls, haxmUrl, Platform.getName());
+    } catch (IOException e) {
+      e.printStackTrace();
+    } catch (XPathException e) {
+      e.printStackTrace();
+    } catch (ParserConfigurationException e) {
+      e.printStackTrace();
+    } catch (SAXException e) {
+      e.printStackTrace();
+    }
+    createInitLayout(downloadUrls);
+    setAlwaysOnTop(true);
+    setVisible(true);
   }
   
-  public void run() {
+  public void run(SDKUrlHolder downloadUrls, File sdkFolder) {
     cancelled = false;
-    downloadTask = new SDKDownloadTask();
+    downloadTask = new SDKDownloadTask(downloadUrls,sdkFolder);
     downloadTask.addPropertyChangeListener(this);
     downloadTask.execute();
-    setAlwaysOnTop(true);
```

Comment:
I think these two lines could resolve the problem of modal window always on top: `setAlwaysOnTop(true);` and `setVisible(true);` Toggling these properties to false when window needs to be hidden. It is worth to test unless your research indicates there is a better way to do this. @DeerajTheepshi

Reply:
Yeah, this was my initial solution , but there is an issue with this. If you try implementing setAlwaysOnTop(false) to the dialog: I experienced this -> When switching to another window, say chrome, as expected the window is not visible. But now when I retry to open processing window via the task bar, processing PDE is not opened but only the dialog is opened. Only after closing the sdk download window, PDE becomes visible again . (Tested on : Win10 x86).

Reply:
I see. Yes, this could be capture in the bug about modal window. This is important information and an interesting challenge to find the right solution, or the most convenient one. Feel free to mark as resolve. @DeerajTheepshi